src/slice.rs

```diff
@@ -0,0 +1,33 @@
+use crate::display_list::DisplayList;
+use std::fmt;
+
+#[derive(Debug, Clone, Default)]
+pub struct Slice<'s> {
+    pub source: &'s str,
```

Comment:
In one of my projects I'm generating this from a `Path`, so I'd need to allocate anyway. Kinda. We could make it generic over `Display`, but that would probably end up spreading generic parameters everywhere. I can allocate everything beforehand and just pass the reference in here, so it's not a blocker. Just wanted to comment about my use case.